### ISSUE TITLE
`-n` styling command fails with file-to-file comparison

### ISSUE THREAD

**Issue by tillig** [pb33f/openapi-changes#29]
Running `openapi-changes summary` to compare two files works great:

```sh
openapi-changes summary ./schema.json ./test-schema.json
```

In tests, this yields:

```text
@@@@@@@   @@@@@@@   @@@@@@   @@@@@@   @@@@@@@@
@@@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@@  @@@@@@@@
@@!  @@@  @@!  @@@      @@@      @@@  @@!
!@!  @!@  !@   @!@      @!@      @!@  !@!
@!@@!@!   @!@!@!@   @!@!!@   @!@!!@   @!!!:!
!!@!!!    !!!@!!!!  !!@!@!   !!@!@!   !!!!!:
!!:       !!:  !!!      !!:      !!:  !!:
:!:       :!:  !:!      :!:      :!:  :!:
 ::        :: ::::  :: ::::  :: ::::   ::
 :        :: : ::    : : :    : : :    :


openapi-changes version: 0.0.20 | compiled: 2023-03-30T14:51:42Z


 INFO  printing summary
└─┬Paths
  └─┬/banking/v1/products
    └─┬GET
      └─┬Responses
        └─┬200
          └─┬application/json
            └─┬Schema
              └─┬Schema
                └──[M] $ref (180:18)❌

Date: 04/14/23 | Commit: New: ./test-schema.json, Original: ./schema.json
Document Element | Total Changes | Breaking Changes
paths            | 1             | 1
❌ 1 Breaking changes out of 1

  ERROR   breaking changes discovered
```

If I add the `-n` parameter to turn off coloring, it seems to want to try doing a git-based diff and ignores the files.

```sh
openapi-changes summary -n ./schema.json ./test-schema.json
```

This yields...

```text
@@@@@@@   @@@@@@@   @@@@@@   @@@@@@   @@@@@@@@
@@@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@@  @@@@@@@@
@@!  @@@  @@!  @@@      @@@      @@@  @@!
!@!  @!@  !@   @!@      @!@      @!@  !@!
@!@@!@!   @!@!@!@   @!@!!@   @!@!!@   @!!!:!
!!@!!!    !!!@!!!!  !!@!@!   !!@!@!   !!!!!:
!!:       !!:  !!!      !!:      !!:  !!:
:!:       :!:  !:!      :!:      :!:  :!:
 ::        :: ::::  :: ::::  :: ::::   ::
 :        :: : ::    : : :    : : :    :


openapi-changes version: 0.0.20 | compiled: 2023-03-30T14:51:42Z

starting work.
Building original model for commit 04dd12
```

...and then it hangs, for minutes. It never completes so I have to ^C kill it.

**Comment by daveshanley** [comment]
Whoops, this looks like a bug. Thanks for finding it. I will sweep through soon with a fix.

**Comment by OllieTho** [comment]
To add to this, I recently encountered similar behavior with the command
`openapi-changes report -n <file_a> <file_b>`

Without the -n will return results, but adding it causes the ending result to return nothing.

**Comment by tillig** [comment]
This version did allow the comparison to complete, so that's awesome! Thanks!

I did notice that `INFO` messages don't appear if coloring is turned off.

```text
pwsh> openapi-changes summary ./schema.json ./test-schema.json

@@@@@@@   @@@@@@@   @@@@@@   @@@@@@   @@@@@@@@
@@@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@@  @@@@@@@@
@@!  @@@  @@!  @@@      @@@      @@@  @@!
!@!  @!@  !@   @!@      @!@      @!@  !@!
@!@@!@!   @!@!@!@   @!@!!@   @!@!!@   @!!!:!
!!@!!!    !!!@!!!!  !!@!@!   !!@!@!   !!!!!:
!!:       !!:  !!!      !!:      !!:  !!:
:!:       :!:  !:!      :!:      :!:  :!:
 ::        :: ::::  :: ::::  :: ::::   ::
 :        :: : ::    : : :    : : :    :


openapi-changes version: 0.0.21 | compiled: 2023-04-20T15:53:45Z


 INFO  printing summary
pwsh> openapi-changes summary -n ./schema.json ./test-schema.json

@@@@@@@   @@@@@@@   @@@@@@   @@@@@@   @@@@@@@@
@@@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@@  @@@@@@@@
@@!  @@@  @@!  @@@      @@@      @@@  @@!
!@!  @!@  !@   @!@      @!@      @!@  !@!
@!@@!@!   @!@!@!@   @!@!!@   @!@!!@   @!!!:!
!!@!!!    !!!@!!!!  !!@!@!   !!@!@!   !!!!!:
!!:       !!:  !!!      !!:      !!:  !!:
:!:       :!:  !:!      :!:      :!:  :!:
 ::        :: ::::  :: ::::  :: ::::   ::
 :        :: : ::    : : :    : : :    :


openapi-changes version: 0.0.21 | compiled: 2023-04-20T15:53:45Z


pwsh>
```

Is that the correct behavior?

Also, is there a flag to turn the big header off?

**Comment by OllieTho** [comment]
> The reason why you see nothing when using `-n` with the `report` command, is because the `-n` flag is not supported by that function (it outputs nothing but JSON).

If providing an invalid OAS specification, the error returned by the `report` command will have stylings.
![openapichanges-report-output](https://user-images.githubusercontent.com/117213665/233699223-c1173837-1439-4a24-804a-69a8cc800e00.PNG)

In this case I had a `parameters` attribute that did not have the included parameters in an array.
Turning off the command line stylings in this case is important so we can view these errors clearly in our CI/CD pipeline.

**Comment by daveshanley** [comment]
I see, I missed this edgecase, another fix inbound.

**Comment by daveshanley** [comment]
> This version did allow the comparison to complete, so that's awesome! Thanks!
> 
> I did notice that `INFO` messages don't appear if coloring is turned off.
> 
> ```
> pwsh> openapi-changes summary ./schema.json ./test-schema.json
> 
> @@@@@@@   @@@@@@@   @@@@@@   @@@@@@   @@@@@@@@
> @@@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@@  @@@@@@@@
> @@!  @@@  @@!  @@@      @@@      @@@  @@!
> !@!  @!@  !@   @!@      @!@      @!@  !@!
> @!@@!@!   @!@!@!@   @!@!!@   @!@!!@   @!!!:!
> !!@!!!    !!!@!!!!  !!@!@!   !!@!@!   !!!!!:
> !!:       !!:  !!!      !!:      !!:  !!:
> :!:       :!:  !:!      :!:      :!:  :!:
>  ::        :: ::::  :: ::::  :: ::::   ::
>  :        :: : ::    : : :    : : :    :
> 
> 
> openapi-changes version: 0.0.21 | compiled: 2023-04-20T15:53:45Z
> 
> 
>  INFO  printing summary
> pwsh> openapi-changes summary -n ./schema.json ./test-schema.json
> 
> @@@@@@@   @@@@@@@   @@@@@@   @@@@@@   @@@@@@@@
> @@@@@@@@  @@@@@@@@  @@@@@@@  @@@@@@@  @@@@@@@@
> @@!  @@@  @@!  @@@      @@@      @@@  @@!
> !@!  @!@  !@   @!@      @!@      @!@  !@!
> @!@@!@!   @!@!@!@   @!@!!@   @!@!!@   @!!!:!
> !!@!!!    !!!@!!!!  !!@!@!   !!@!@!   !!!!!:
> !!:       !!:  !!!      !!:      !!:  !!:
> :!:       :!:  !:!      :!:      :!:  :!:
>  ::        :: ::::  :: ::::  :: ::::   ::
>  :        :: : ::    : : :    : : :    :
> 
> 
> openapi-changes version: 0.0.21 | compiled: 2023-04-20T15:53:45Z
> 
> 
> pwsh>
> ```
> 
> Is that the correct behavior?
> 
> Also, is there a flag to turn the big header off?

Can you add a feature request to add a new flag to turn off the banner?

**Comment by OllieTho** [comment]
> I see, I missed this edgecase, another fix inbound.

Hey, just wanted to check in and see if there were any status updates on this! Thanks!

**Comment by daveshanley** [comment]
Can you try `v0.0.22`?

**Comment by daveshanley** [comment]
The `report` command now supports `-n`

**Comment by OllieTho** [comment]
> Can you try `v0.0.22`?

When I run the report command on v0.0.22 with or without the `-n` option, I'm getting a deadlock error:
```
fatal error: all goroutines are asleep - deadlock!

goroutine 1 [chan send]:
github.com/pb33f/openapi-changes/model.SendProgressError(...)
        /home/runner/work/openapi-changes/openapi-changes/model/progress.go:43
github.com/pb33f/openapi-changes/git.BuildCommitChangelog({0xc000cf7b28, 0x2, 0xc000127b38?}, 0x1?, 0x1?)
        /home/runner/work/openapi-changes/openapi-changes/git/read_local.go:167 +0xc45
github.com/pb33f/openapi-changes/cmd.runLeftRightReport({0xc00001e450, 0x21}, {0xc0000221e0, 0x1e}, 0x0?, 0x0?)
        /home/runner/work/openapi-changes/openapi-changes/cmd/report.go:305 +0x5fb
github.com/pb33f/openapi-changes/cmd.GetReportCommand.func1(0xc000004900?, {0xc0000ac300, 0x2, 0x3?})
        /home/runner/work/openapi-changes/openapi-changes/cmd/report.go:168 +0x68b
github.com/spf13/cobra.(*Command).execute(0xc000004900, {0xc0000ac2a0, 0x3, 0x3})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x12097e0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/pb33f/openapi-changes/cmd.Execute({0xe0ae94?, 0x0?}, {0xe11680?, 0x0?}, {0xe0cf50?, 0xc000060000?})
        /home/runner/work/openapi-changes/openapi-changes/cmd/root.go:48 +0xaa
main.main()
        /home/runner/work/openapi-changes/openapi-changes/openapi-changes.go:26 +0x10e

goroutine 18 [chan send]:
github.com/pb33f/openapi-changes/cmd.GetReportCommand.func1.1(0xc00009e060, 0xc00009e0c0)
        /home/runner/work/openapi-changes/openapi-changes/cmd/report.go:68 +0xd0
created by github.com/pb33f/openapi-changes/cmd.GetReportCommand.func1
        /home/runner/work/openapi-changes/openapi-changes/cmd/report.go:167 +0x65c

goroutine 116 [chan send]:
github.com/pb33f/libopenapi/datamodel/low/v3.(*Paths).Build.func1(0xc0006a92c0, 0x0?, 0xc00066cf90?, 0x904?)
        /home/runner/go/pkg/mod/github.com/pb33f/libopenapi@v0.8.0/datamodel/low/v3/paths.go:107 +0x372
created by github.com/pb33f/libopenapi/datamodel/low/v3.(*Paths).Build
        /home/runner/go/pkg/mod/github.com/pb33f/libopenapi@v0.8.0/datamodel/low/v3/paths.go:134 +0x2b8
```

The last block in the error is repeated a few dozen times as well.

**Comment by daveshanley** [comment]
I think this may have been fixed with `v0.0.23` can you let me know, I can't re-create this anymore.

**Comment by OllieTho** [comment]
Looks like it's resolved and working great! Thanks!
